internal/gps/cmd.go

```diff
@@ -196,12 +198,31 @@ func runFromRepoDir(ctx context.Context, repo vcs.Repo, timeout time.Duration, c
 	return c.combinedOutput(ctx)
 }
 
+func getenvDuration(key string, def time.Duration) time.Duration {
+	seconds, err := strconv.Atoi(os.Getenv(key))
+
+	if err != nil {
+		return def
+	}
+
+	return time.Duration(seconds) * time.Second
+}
+
 const (
+	// envExpensiveCmdTimeout is the environment variable to read for the
+	// expensiveCmdTimeout default, in seconds.
+	envExpensiveCmdTimeout = "DEP_EXPENSIVE_CMD_TIMEOUT"
```

Comment:
I actually felt uncomfortable adding those environment variables where no other were currently used.

I went with those anyway because:
- Timeouts are not something I'd like to set on each invocation of `dep`. If an operation takes a long time on my machine, this is likely to happen often - or even every time.
- It somehow felt like leaking an implementation detail: if we end up having 15 sorts of timeouts in `dep` and all of those have to be exposed as flags (part of the CLI API so to speak), things will get ugly fast.

That being said, when in Rome, I'll do as the romans do so feel free to confirm or reject my decisions and I'll happily update my PR !

`dep` is awesome :)